src/main/java/org/bitsofinfo/hazelcast/discovery/consul/ConsulClientBuilder.java

```diff
@@ -0,0 +1,126 @@
+package org.bitsofinfo.hazelcast.discovery.consul;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.ws.rs.client.ClientBuilder;
+
+import com.google.common.net.HostAndPort;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.orbitz.consul.Consul;
+import com.orbitz.consul.Consul.Builder;
+/**
+ * An implementation of a Consul client builder.
+ * 
+ * @author bmudda
+ *
+ */
+public class ConsulClientBuilder implements ConsulBuilder {
+
+	private static final ILogger logger = Logger.getLogger(ConsulClientBuilder.class);	
+	
+	@Override
+	public Consul buildConsul(
+			String consulHost,
+			Integer consulPort,
+			boolean consulSslEnabled,
+			String	consulSslServerCertFilePath,
+			String consulSslServerCertBase64,
+			boolean consulServerHostnameVerify
+			) throws Exception {
+		
+		
+		try{
+			Builder consulBuilder = Consul.builder();
+			
+			//If SSL is enabled via xml configuration, then we use SSL context to build our client
+			if(consulSslEnabled){
+				consulBuilder.withUrl("https://"+consulHost+":"+consulPort)
```

Comment:
This does not account for sslEnabled with NO custom cert to trust. I.E someone wanting to use ssl against a Consul node presenting a legit CA signed cert